assets/src/hooks/useVehicles.ts

```diff
@@ -97,12 +147,89 @@ const reducer = (state: State, action: Action): State => {
   }
 }
 
+const scheduleAdherenceStatus = (
+  scheduleAdherenceSecs: number
+): ScheduleAdherenceStatus => {
+  const oneMinuteInSeconds = 60
+  const sixMinutesInSeconds = 360
+
+  if (scheduleAdherenceSecs < -oneMinuteInSeconds) {
+    return "early"
+  } else if (scheduleAdherenceSecs > sixMinutesInSeconds) {
+    return "late"
+  } else {
+    return "on-time"
+  }
+}
+
+const vehicleStopStatusFromData = (
+  vehicleStopStatusData: VehicleStopStatusData
+): VehicleStopStatus => ({
+  status: vehicleStopStatusData.status,
+  stopId: vehicleStopStatusData.stop_id,
+  stopName: vehicleStopStatusData.stop_name,
+})
+
+const vehicleTimepointStatusFromData = (
+  vehicleTimepointStatusData: VehicleTimepointStatusData | null
+): VehicleTimepointStatus | null => {
+  if (vehicleTimepointStatusData) {
+    return {
+      fractionUntilTimepoint:
+        vehicleTimepointStatusData.fraction_until_timepoint,
+      timepointId: vehicleTimepointStatusData.timepoint_id,
+    }
+  } else {
+    return null
+  }
+}
+
+const vehicleFromData = (vehicleData: VehicleData): Vehicle => ({
+  id: vehicleData.id,
+  label: vehicleData.label,
+  runId: vehicleData.run_id,
```

Comment:
Oh thank you for doing this. Having `snake_case` scattered throughout the javascript was bugging me.